browser/ui/views/split_view/split_view_browsertest.cc

```diff
@@ -1164,3 +1169,109 @@ IN_PROC_BROWSER_TEST_F(SplitViewBrowserTest, SplitViewFullscreenTest) {
 }
+
+class SplitViewCookieBrowserTest : public SplitViewBrowserTest {
+ public:
+  void SetUpOnMainThread() override {
+    SplitViewBrowserTest::SetUpOnMainThread();
+    embedded_test_server()->RegisterRequestHandler(base::BindRepeating(
+        [](const net::test_server::HttpRequest& request)
+            -> std::unique_ptr<net::test_server::HttpResponse> {
+          auto response =
+              std::make_unique<net::test_server::BasicHttpResponse>();
+          response->set_content(request.headers.count("Cookie")
+                                    ? request.headers.at("Cookie")
+                                    : "NONE");
+          response->set_code(net::HTTP_OK);
+          return response;
+        }));
+
+    ASSERT_TRUE(embedded_test_server()->Start());
+  }
+
+  void TearDownOnMainThread() override {
+    ASSERT_TRUE(embedded_test_server()->ShutdownAndWaitUntilComplete());
+    SplitViewBrowserTest::TearDownOnMainThread();
+  }
+};
+
+class MockBraveRenderViewContextMenu : public BraveRenderViewContextMenu {
+ public:
+  MockBraveRenderViewContextMenu(content::RenderFrameHost& render_frame_host,
+                                 const content::ContextMenuParams& params)
+      : BraveRenderViewContextMenu(render_frame_host, params) {}
+  ~MockBraveRenderViewContextMenu() override = default;
+
+  void Show() override {}
+};
+
+IN_PROC_BROWSER_TEST_F(SplitViewCookieBrowserTest, CookieTest) {
+  // Navigate to a page and sets a cookie.
+  GURL target_url = embedded_test_server()->GetURL("/");
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), target_url));
+  content::WebContents* web_contents =
+      browser()->tab_strip_model()->GetActiveWebContents();
+  ASSERT_TRUE(
+      content::ExecJs(web_contents->GetPrimaryMainFrame(),
+                      "document.cookie = 'Strict=value1; SameSite=Strict;'"));
+  ASSERT_TRUE(content::ExecJs(web_contents->GetPrimaryMainFrame(),
+                              "document.cookie = 'Lax=value2; SameSite=Lax;'"));
+
+  // Navigate to a page that sub page to check cookies.
+  target_url = embedded_test_server()->GetURL("/sub");
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), target_url));
+  auto cookie =
+      content::EvalJs(web_contents->GetPrimaryMainFrame(), "document.cookie")
+          .ExtractString();
+  EXPECT_EQ(cookie, "Strict=value1; Lax=value2");
+
+  // Open up different site
+  chrome::AddTabAt(browser(), GURL("about:blank"), /*index*/ 1,
+                   /*foreground*/ true);
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), GURL("about:blank")));
+
+  // Triggers "Open link in split view" context menu command.
+  auto* initiator_contents =
+      browser()->tab_strip_model()->GetActiveWebContents();
+
+  content::ContextMenuParams params;
+  params.page_url = initiator_contents->GetLastCommittedURL();
+  params.frame_url =
+      initiator_contents->GetPrimaryMainFrame()->GetLastCommittedURL();
```

Comment:
shouldn't this be the URL of the iframe (not main frame) if the user is opening a link that's inside an iframe?

Reply:
might be worth adding a test case for that (link opened via context menu inside an iframe)

Reply:
I'm not sure if it's necessary here, as it's mocking the STR that was reported. this test page doesn't have any subframe.

Reply:
Could you tell me what is expected behavior in that case?

Reply:
@sangwoo108 frame origin should be the iframe's origin, not the origin of the main frame.

Reply:
@diracdeltas , could you illustrate a little bit more with code? I'm not getting what you mean. I understand that when context menu was triggered from subframe frame origin should be the subframe's origin. But in this test case, it is my understanding that we don't have subframe, so context menu's frame origin should be main frame's origin.

Reply:
i was hoping you could add a new test case with a subframe, but @stoletheminerals manually tested the case i was asking about and it works as expected.